packages/pod-install/src/index.ts

```diff
@@ -41,6 +42,26 @@ async function runAsync(): Promise<void> {
 
   const possibleProjectRoot = CocoaPodsPackageManager.getPodProjectRoot(projectRoot);
   if (!possibleProjectRoot) {
+    try {
+      const jsonData = JSON.parse(readFileSync(join(projectRoot, 'package.json')).toString());
+      const hasExpo = jsonData.dependencies?.hasOwnProperty('expo');
+      const hasReactNativeUnimodules = jsonData.dependencies?.hasOwnProperty(
+        'react-native-unimodules'
+      );
+      const isManaged = hasExpo && !hasReactNativeUnimodules;
+
+      if (isManaged) {
+        info(
+          chalk.yellow(
+            'In a managed project, `pod-install` will run automatically as part of the EAS Build process. Learn more: https://docs.expo.dev/build-reference/ios-builds/'
+          )
+        );
+        return;
+      }
+    } catch {
+      // Expected to throw if no package.json is present
```

Comment:
we should check if package.json exists rather than depending on try/catch for this, otherwise we'll unintentionally swallow any other exceptions that may arise from the above block